prestashop8.x/tawkto.php

```diff
@@ -281,4 +291,47 @@ private function getArrayFromJson($data)
 
         return $arr;
     }
+
+    /**
+     * Retrieve JS API key
+     *
+     * @param string $js_api_key Encrypted JS API key
+     *
+     * @return string
+     */
+    private function getJsApiKey(string $js_api_key)
+    {
+        // Cache::store & Cache::retrieve are not persistent
+
+        $key = $this->getDecryptedData($js_api_key);
+
+        return $key;
+    }
+
+    /**
+     * Decrypt data
+     *
+     * @param string $data Data to decrypt
+     *
+     * @return string
+     */
+    private function getDecryptedData(string $data)
+    {
+        $decoded = base64_decode($data);
+
+        if ($decoded === false) {
+            return '';
```

Comment:
Avoid returning empty strings on failures. It could cause hard-to-debug behavior. If anything - error case should be handled where the de-crypted [API key is used](https://github.com/tawk/tawk-prestashop/pull/45/files#diff-cd3bec509cc47feb91bd7296e15bf72d43433e564d74020e05d1be2ccd59ae2bR179)